src/Hangfire.EmailNotification/ClientNotifiers/SmtpClientNotifier.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mail;
+using System.Net.Mime;
+using System.Text;
+using System.Threading.Tasks;
+using Hangfire.Notification;
+
+namespace Hangfire.Notification
+{
+    public class SmtpClientNotifier : INotifier
+    {
+        private SmtpClient _smtpClient;
+        Dictionary<EventTypes.Events, List<string>> _eventReceivers = new Dictionary<EventTypes.Events, List<string>>();
+        private string _fromEmail;
+        private string _fromName;
+        
+        public SmtpClientNotifier(string fromEmail, string fromName, SmtpConfiguration sendgridConfig)
+        {
+            _fromEmail = fromEmail;
+            _fromName = fromName;
+            _smtpClient = new SmtpClient(sendgridConfig.Host, sendgridConfig.Port);
+            _smtpClient.Credentials = new System.Net.NetworkCredential(sendgridConfig.Username, sendgridConfig.Password);
+        }
+
+        public void Notify(EventTypes.Events eventType, string subject, string message)
+        {
+            var mailMsg = new MailMessage();
+
+            List<string> emails;
+            _eventReceivers.TryGetValue(eventType, out emails);
```

Comment:
should do a bool check in the if before the list check as it is possible "emails" could be null